tests/integration-tests/jobs/createJobKubernetes-test.js

```diff
@@ -218,12 +224,150 @@ describe('Create job specific kubernetes tests', async function () {
+                                    start_after: 0
+                                },
+                                {
+                                    experiment_id: chaosExperimentId,
+                                    start_after: 2
+                                }
+                            ]
+                        };
+
+                        createJobResponse = await schedulerRequestCreator.createJob(jobBody, {
+                            'Content-Type': 'application/json'
+                        });
+
+                        should(createJobResponse.status).eql(201);
+                        jobId = createJobResponse.body.id;
+                    });
+
+                    it('use delete all containers', async () => {
+                        nock(kubernetesConfig.kubernetesUrl).get(`/apis/batch/v1/namespaces/${kubernetesConfig.kubernetesNamespace}/jobs?labelSelector=app=predator`)
+                            .reply(200, {
+                                items: [{ metadata: { uid: 'x' } }]
+                            });
+                        nock(kubernetesConfig.kubernetesUrl).get(`/api/v1/namespaces/${kubernetesConfig.kubernetesNamespace}/pods?labelSelector=controller-uid=x`)
+                            .reply(200, {
+                                items: [{ metadata: { name: 'podA' } }]
+                            });
+
+                        nock(kubernetesConfig.kubernetesUrl).get(`/api/v1/namespaces/${kubernetesConfig.kubernetesNamespace}/pods/podA`)
+                            .reply(200, {
+                                metadata: { labels: { 'job-name': 'predator.job' } },
+                                status: {
+                                    containerStatuses: [{
+                                        name: 'predator-runner',
+                                        state: { terminated: { finishedAt: '2020' } }
+                                    }, {
+                                        name: 'podB',
+                                        state: {}
+                                    }]
+                                }
+
+                            });
+
+                        nock(kubernetesConfig.kubernetesUrl).delete(`/apis/batch/v1/namespaces/${kubernetesConfig.kubernetesNamespace}/jobs/predator.job?propagationPolicy=Foreground`)
+                            .reply(200);
+
+                        chaosExperimentsRequestCreator.nockK8sChaosExperimentSupportedKinds();
+
+                        nock(kubernetesConfig.kubernetesUrl).get('/apis/chaos-mesh.org/v1alpha1/podchaos?labelSelector=app=predator')
+                            .reply(200, {
+                                items: [{
+                                    metadata: { name: 'first-exp', namespace: kubernetesConfig.kubernetesNamespace },
+                                    status: {
+                                        experiment: {
+                                            desiredPhase: 'Stop'
```

Comment:
do we test that desiredPhase != stop is not deleted?

do we test failure scenarios - k8s api fails to delete

do we test non app=predator label not deleted (?)

Reply:
1. yes - you can see that I used one with Run and it was not counted as deleted in counter - only 2 deleted.
2. No, but I think this whole scope doesn't check failures from kubernetes, we check it on unit tests - here the purpose is to check the flow.
3. Not really, but we only do get for predator label so we don't really have that info.

Reply:
1 - resolve
2- okish even though it is possible to test
3 resolve

Reply:
For 2 we will have a separated PR for these negative scenarios